cli/cmd/generate_execute.go

```diff
@@ -0,0 +1,392 @@
+	case noop:
+		return "(noop)"
+	case replace:
+		return "replaced"
+	case createBeforeDestroy:
+		return "created before destroy"
+	case destroyBeforeCreate:
+		return "destroyed before create"
+	default:
+		return "unchanged"
+	}
+}
+
+type tfPlanChangesSummary struct {
+	plan    *tfjson.Plan
+	create  int
+	deleted int
+	update  int
+	replace int
+}
+
+// used to display the results of a plan
+//
+// returns true if apply should run, false to exit
+func DisplayTerraformPlanChanges(tf *tfexec.Terraform, data tfPlanChangesSummary) (bool, error) {
+	// Prompt for next steps
+	prompt := true
+	previewShown := false
+	var answer int
+
+	// Displaying resources
+	for prompt {
+		id, err := promptForTerraformNextSteps(&previewShown, data)
+		if err != nil {
+			return false, errors.Wrap(err, "failed to run terraform")
+		}
+
+		switch {
+		case id == 1 && !previewShown:
+			cli.OutputHuman("Resource details: \n")
+			for _, c := range data.plan.ResourceChanges {
+				cli.OutputHuman(fmt.Sprintf("  %s.%s will be %s\n", c.Type, c.Name,
+					createOrDestroy(
+						c.Change.Actions.Create(),
+						c.Change.Actions.Delete(),
+						c.Change.Actions.Update(),
+						c.Change.Actions.Read(),
+						c.Change.Actions.NoOp(),
+						c.Change.Actions.Replace(),
+						c.Change.Actions.CreateBeforeDestroy(),
+						c.Change.Actions.DestroyBeforeCreate(),
+					),
+				),
+				)
+			}
+			cli.OutputHuman("\n")
+			cli.OutputHuman("More details can be viewed by running:\n\n  cd %s\n  %s show tfplan.json\n", tf.WorkingDir(), tf.ExecPath())
+			cli.OutputHuman("\n")
+		default:
+			answer = id
+			prompt = false
+		}
+
+		if id == 1 && !previewShown {
+			previewShown = true
+		}
+	}
+
+	// Run apply
+	if answer == 0 {
+		return true, nil
+	}
+
+	// Quit
+	return false, nil
+}
+
+func processTfPlanChangesSummary(tf *tfexec.Terraform) (*tfPlanChangesSummary, error) {
+	// Extract changes from tf plan
+	cli.StartProgress("Getting terraform plan details")
+	plan, err := tf.ShowPlanFile(context.Background(), "tfplan.json")
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to inspect terraform plan")
+	}
+	cli.StopProgress()
+
+	// Build output of changes
+	resourceCreate := 0
+	resourceDelete := 0
+	resourceUpdate := 0
+	resourceReplace := 0
+
+	for _, c := range plan.ResourceChanges {
+		switch {
+		case c.Change.Actions.Create():
+			resourceCreate += 1
+		case c.Change.Actions.Delete():
+			resourceDelete += 1
+		case c.Change.Actions.Update():
+			resourceUpdate += 1
+		case c.Change.Actions.Replace():
+			resourceReplace += 1
+		}
+	}
+
+	return &tfPlanChangesSummary{
+		plan:    plan,
+		create:  resourceCreate,
+		deleted: resourceDelete,
+		update:  resourceDelete,
+		replace: resourceReplace,
+	}, nil
+}
+
+func TerraformInit(tf *tfexec.Terraform) error {
+	cli.StartProgress("Running terraform init")
+	if err := tf.Init(context.Background(), tfexec.Upgrade(true)); err != nil {
+		return errors.Wrap(err, "failed to execute terraform init")
```

Comment:
nit: this prints on the same line as the progress. Would be nice to have a newline before the error

`\Running terraform init ERROR failed to run terraform apply: failed to execute terraform init: exit status 1`

Reply:
👍